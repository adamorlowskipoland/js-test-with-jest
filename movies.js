export default {
  add(movies, newMovie) {
    movies.push({
      title: newMovie,
      rate: null
    })
  },
  rate(movie, rate) {
    movie.rate = rate
  }
}
