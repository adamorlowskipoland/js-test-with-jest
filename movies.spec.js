import movies from "./movies"

describe("Favourite Movies", () => {
  let myMovies
  beforeEach(() => {
    myMovies = [{
      title: "Ice Age",
      rate: null
    }]
  })
  test("can add a movie", () => {
    movies.add(myMovies, 'Avatar')
    expect(myMovies).toMatchSnapshot()
  })

  test("rate a movie", () => {
    movies.rate(myMovies[0], 5)
    expect(myMovies).toMatchSnapshot()
  })
})
