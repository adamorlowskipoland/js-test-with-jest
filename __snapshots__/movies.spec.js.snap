// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourite Movies can add a movie 1`] = `
Array [
  Object {
    "rate": null,
    "title": "Ice Age",
  },
  Object {
    "rate": null,
    "title": "Avatar",
  },
]
`;

exports[`Favourite Movies rate a movie 1`] = `
Array [
  Object {
    "rate": 5,
    "title": "Ice Age 2",
  },
]
`;
